tests/rptest/tests/redpanda_startup_test.py

```diff
@@ -96,3 +97,78 @@ def check_fips_mode_metric(
             metrics_name='redpanda_application_fips_mode',
             metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS,
             expected_mode=RedpandaServiceBase.FIPSMode.permissive)
+
+    @cluster(num_nodes=3)
+    def test_non_homogenous(self):
+        """
+        This test will validate that the non homogenous FIPS
+        metric responds appropriately
+
+        1. Find the lader
+        2. On a non-leader node, set to 'disabled'
+        3. Restart that node
+        4. Wait for count to go to 1
+        5. On same node, set back to permissive
+        6. Restart node
+        7. Wait for count to go back to 0
+        """
+
+        leader = self.redpanda.controller()
+        non_leader_nodes = list(set(self.redpanda.nodes).difference({leader}))
+        assert len(non_leader_nodes) > 0, "Could not find any non-leader nodes"
+        target_node = non_leader_nodes[0]
+        self.logger.debug(f'Leader: {leader}')
+        self.logger.debug(f'Non-leader: {target_node}')
+        NON_FIPS_CONFIG = dict(fips_mode="disabled",
+                               openssl_config_file=None,
+                               openssl_module_directory=None)
+        FIPS_CONFIG = dict(
+            fips_mode="permissive",
+            openssl_config_file=self.redpanda.get_openssl_config_file_path(),
+            openssl_module_directory=self.redpanda.
+            get_openssl_modules_directory())
+
+        def check_non_homogenous_value(metrics_name: str,
+                                       metrics_endpoint: MetricsEndpoint,
+                                       leader_node: ClusterNode,
+                                       expected_val: int):
+            metrics = self.redpanda.metrics_sample(
+                sample_pattern=metrics_name, metrics_endpoint=metrics_endpoint)
+            assert isinstance(
+                metrics,
+                MetricSamples), f'Failed to get metrics {metrics_name}'
+
+            # Should only get one
+            samples = [
+                sample for sample in metrics.samples
+                if sample.node == leader_node
+            ]
+            assert len(
+                samples) == 1, f'Invalid number of samples: {len(samples)}'
+            assert int(
+                samples[0].value
+            ) == expected_val, f'Unexpected count: {expected_val} != {int(samples[0].value)}'
+
+        check_non_homogenous_value(
+            metrics_name='redpanda_cluster_non_homogenous_fips_mode',
+            metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS,
+            leader_node=leader,
+            expected_val=0)
+
+        self.redpanda.restart_nodes([target_node],
+                                    override_cfg_params=NON_FIPS_CONFIG)
+
+        check_non_homogenous_value(
+            metrics_name='redpanda_cluster_non_homogenous_fips_mode',
+            metrics_endpoint=MetricsEndpoint.PUBLIC_METRICS,
+            leader_node=leader,
+            expected_val=1)
```

Comment:
q: should this be something we wait_until? my intuition is "no" since we're just reading the node config, but I figured I'd raise it anyway.